flash/image/classification/data.py

```diff
@@ -252,6 +611,114 @@ def from_csv(
+
+        Input images will be extracted from the ``input_field`` column in the CSV files.
+        The supported file extensions are: ``.jpg``, ``.jpeg``, ``.png``, ``.ppm``, ``.bmp``, ``.pgm``, ``.tif``,
+        ``.tiff``, ``.webp``, and ``.npy``.
+        The targets will be extracted from the ``target_fields`` in the CSV files and can be in any of our
+        :ref:`supported classification target formats <formatting_classification_targets>`.
+        To learn how to customize the transforms applied for each stage, read our
+        :ref:`customizing transforms guide <customizing_transforms>`.
+
+        Args:
+            input_field: The field in the CSV files containing the image files.
+            target_fields: The field or list of fields in the CSV files containing the targets.
+            train_file: The CSV file to use when training.
+            train_images_root: The root directory containing train images.
+            train_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            val_file: The CSV file to use when validating.
+            val_images_root: The root directory containing validation images.
+            val_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            test_file: The CSV file to use when testing.
+            test_images_root: The root directory containing test images.
+            test_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            predict_file: The CSV file to use when predicting.
+            predict_images_root: The root directory containing predict images.
+            predict_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an
+                image file path.
+            train_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when training.
+            val_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when validating.
+            test_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when testing.
+            predict_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when
+              predicting.
+            input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the data.
+            transform_kwargs: Dict of keyword arguments to be provided when instantiating the transforms.
+            data_module_kwargs: Additional keyword arguments to provide to the
+              :class:`~flash.core.data.data_module.DataModule` constructor.
+
+        Returns:
+            The constructed :class:`~flash.image.classification.data.ImageClassificationData`.
+
+        Example::
+
+            .. testsetup::
```

Comment:
Could this be added to a private file _data_doc.py as this could be used to generate more complex data.

Reply:
>>> from flash.image.classification.data_doc import test_from_folders
>>> test_from_folders()

Just an idea :)

Reply:
Yeah, we could do this, I think eventually they will come from `tests/helpers`. It doesn't matter too much as the contents of the `testsetup` block will be hidden in the rendered docs so never seen by users.